src/odemis/driver/simstreakcam.py

```diff
@@ -0,0 +1,537 @@
+        resolution = self._shape[:2]
+        new_res = (int(resolution[0] // self._binning[0]),
+                    int(resolution[1] // self._binning[1]))  # floor division
+
+        if self._spectrograph:
+            self._updateWavelengthList()  # update WavelengthList when changing binning
+
+        return new_res
+
+    def _setCamExpTime(self, value):
+        """
+        Set the camera exposure time.
+        :parameter value: (float) exposure time to be set
+        :return: (float) current exposure time
+        """
+        self._metadata[model.MD_EXP_TIME] = value  # update MD
+
+        return value
+
+    def _sync(self, event):  # event = self.softwareTrigger
+        """
+        Synchronize the acquisition on the given event. Every time the event is
+          triggered, the DataFlow will start a new acquisition.
+        Behaviour is unspecified if the acquisition is already running.  # TODO still True?
+        event (model.Event or None): event to synchronize with. Use None to
+          disable synchronization.
+        The DataFlow can be synchronized only with one Event at a time.
+        """
+        # if event None and sync as well -> return, or if event sync, but sync already set -> return
+        if self._sync_event == event:
+            return
+
+        if self._sync_event:  # if new event = None, unsubscribe previous event (which was softwareTrigger)
+            self._sync_event.unsubscribe(self)
+            if not event:
+                self._evtq.put(None)  # in case it was waiting for this event
+
+        self._sync_event = event
+
+        if self._sync_event:
+            # softwareTrigger subscribes to onEvent method: if softwareTrigger.notify() called, onEvent method called
+            self._sync_event.subscribe(self)  # must have onEvent method
+            self._evtq = Queue.Queue()  # to be sure it's empty
+
+    @oneway
+    def onEvent(self):
+        """
+        Called by the Event when it is triggered  (e.g. self.softwareTrigger.notify()).
+        """
+        logging.debug("Event triggered to start a new synchronized acquisition.")
+        self._evtq.put(time.time())
+
+    # override
+    def updateMetadata(self, md):
+        """
+        Update the metadata.
+        """
+        super(ReadoutCamera, self).updateMetadata(md)
+        if model.MD_LENS_MAG in md:
+            self._updateWavelengthList()
+
+    def _mergeMetadata(self, md):
+        """
+        Create dict containing all metadata from the children readout camera, streak unit,
+        delay genereator and the metadata from the parent streak camera.
+        :return: merged metadata
+        """
+
+        md_devices = [self.parent._streakunit._metadata, self.parent._delaybox._metadata]
+
+        for md_dev in md_devices:
+            for key in md_dev.keys():
+                if key not in md.keys():
+                    md[key] = md_dev[key]
+                else:
```

Comment:
That's a trick that only works for strings. I'd suggest to just hardcode the metadata which can work with this. So something like:
`elif md in (MD_HW_NAME, MD_HW_VERSION, MD_SW_VERSION):`